### ISSUE TITLE
org.graalvm.compiler.java.BytecodeParser$BytecodeParserError: java.lang.NullPointerException with SQLite

### ISSUE THREAD

**Issue by martinhutton** [quarkusio/quarkus#27271]
### Describe the bug

Small application using Quarkus and SQLite causes a NPE in `org.graalvm.compiler.java.BytecodeParser ` and native application cannot be built.

This occurs on Windows 10. I found similar issues for Mac and Linux but they appear to have been resolved now.

```
C:\work\quarkus-sqlite-test>gradlew build -Dquarkus.package.type=native
Uber JAR strategy is used for native image source JAR generation on Windows. This is done for the time being to work around a current GraalVM limitation on Windows concerning the maximum command length (see https://github.com/oracle/graal/issues/2387).
Dependencies with duplicate files detected. The dependencies [org.glassfish:jakarta.el::jar:3.0.4[paths: C:\Users\Martyn\.gradle\caches\modules-2\files-2.1\org.glassfish\jakarta.el\3.0.4\f48473482c0e3e714f87186d9305bcae30b7f5cb\jakarta.el-3.0.4.jar;], jakarta.el:jakarta.el-api::jar:3.0.3[paths: C:\Users\Martyn\.gradle\caches\modules-2\files-2.1\jakarta.el\jakarta.el-api\3.0.3\f311ab94bb1d4380690a53d737226a6b879dd4f1\jakarta.el-api-3.0.3.jar;]] contain duplicate files, e.g. javax/el/StaticFieldELResolver.class
Dependencies with duplicate files detected. The dependencies [com.sun.activation:jakarta.activation::jar:1.2.1[paths: C:\Users\Martyn\.gradle\caches\modules-2\files-2.1\com.sun.activation\jakarta.activation\1.2.1\8013606426a73d8ba6b568370877251e91a38b89\jakarta.activation-1.2.1.jar;], javax.activation:javax.activation-api::jar:1.2.0[paths: C:\Users\Martyn\.gradle\caches\modules-2\files-2.1\javax.activation\javax.activation-api\1.2.0\85262acf3ca9816f9537ca47d5adeabaead7cb16\javax.activation-api-1.2.0.jar;]] contain duplicate files, e.g. javax/activation/ActivationDataFlavor.class
========================================================================================================================
GraalVM Native Image: Generating 'quarkus-sqlite-1.0-SNAPSHOT-runner'...
========================================================================================================================
[1/7] Initializing...                                                                                    (8.4s @ 0.26GB)
 Version info: 'GraalVM 22.0.0.2 Java 11 CE'
 3 user-provided feature(s)
  - io.quarkus.runner.AutoFeature
  - io.quarkus.runtime.graal.DisableLoggingAutoFeature
  - io.quarkus.runtime.graal.ResourcesFeature
[2/7] Performing analysis...  [********]                                                                (41.2s @ 3.98GB)
17:52:20,488 INFO  [org.joo.Constants]

jOOQ tip of the day: You don't have to register your Converters or Bindings with the code generator. You can use ad-hoc converters on any Field expression: https://www.jooq.org/doc/latest/manual/sql-execution/fetching/ad-hoc-converter/

  18,982 (90.33%) of 21,015 classes reachable
  30,737 (68.62%) of 44,796 fields reachable
 104,829 (54.96%) of 190,731 methods reachable
   1,046 classes,   521 fields, and 6,198 methods registered for reflection
      94 classes,   104 fields, and   292 methods registered for JNI access
[3/7] Building universe...                                                                               (2.8s @ 1.85GB)
[4/7] Parsing methods...      [***]                                                                      (5.1s @ 1.97GB)

Fatal error: org.graalvm.compiler.java.BytecodeParser$BytecodeParserError: java.lang.NullPointerException
        at parsing org.sqlite.BusyHandler.<init>(BusyHandler.java:10)
        at method: void org.sqlite.BusyHandler.<init>()
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.throwParserError(BytecodeParser.java:2660)
        at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.throwParserError(SharedGraphBuilderPhase.java:106)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3521)
        at com.oracle.svm.hosted.phases.HostedBytecodeParser.iterateBytecodesForBlock(HostedGraphBuilderPhase.java:196)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3473)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3318)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1149)
        at com.oracle.svm.hosted.phases.HostedBytecodeParser.build(HostedGraphBuilderPhase.java:140)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1041)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:84)
        at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:80)
        at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.run(Phase.java:49)
        at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:214)
------------------------------------------------------------------------------------------------------------------------
        at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
        at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
        at com.oracle.svm.hosted.code.CompileQueue.defaultParseFunction(CompileQueue.java:1030)
        at com.oracle.svm.hosted.code.CompileQueue.doParse(CompileQueue.java:762)
        at com.oracle.svm.hosted.code.CompileQueue$ParseTask.run(CompileQueue.java:337)
        at com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:195)
        at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:179)
        at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NullPointerException
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.java.RegisterFinalizerNode.mayHaveFinalizer(RegisterFinalizerNode.java:105)
        at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.StandardGraphBuilderPlugins$34.apply(StandardGraphBuilderPlugins.java:996)
        at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.graphbuilderconf.InvocationPlugin.execute(InvocationPlugin.java:168)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.applyInvocationPlugin(BytecodeParser.java:2261)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.tryInvocationPlugin(BytecodeParser.java:2201)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.appendInvoke(BytecodeParser.java:1886)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genInvokeSpecial(BytecodeParser.java:1775)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genInvokeSpecial(BytecodeParser.java:1767)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5408)
        at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3513)
        ... 23 more
                        6.9s (10.6% of total time) in 30 GCs | Peak RSS: 6.74GB | CPU load: 4.62
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 C:\work\quarkus-sqlite-test\build\quarkus-sqlite-1.0-SNAPSHOT-native-image-source-jar\quarkus-sqlite-1.0-SNAPSHOT-runner.build_artifacts.txt
========================================================================================================================
Failed generating 'quarkus-sqlite-1.0-SNAPSHOT-runner' after 1m 4s.
Error: Image build request failed with exit status 1
> Task :quarkusBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusBuild'.
> io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: io.quarkus.deployment.pkg.steps.NativeImageBuildStep$ImageGenerationFailureException: Image generation failed. Exit code: 1
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.imageGenerationFailed(NativeImageBuildStep.java:400)
        at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:240)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:944)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at java.base/java.lang.Thread.run(Thread.java:829)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)


* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 29s
6 actionable tasks: 6 executed
```

### Expected behavior

Native application

### Actual behavior

NullPointerException thrown

### How to Reproduce?

Clone https://github.com/martinhutton/quarkus-sqlite-test

Run `gradlew build -Dquarkus.package.type=native`

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.19044.1889]

### Output of `java -version`

openjdk version "11.0.14" 2022-01-18 OpenJDK Runtime Environment GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05) OpenJDK 64-Bit Server VM GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05, mixed mode, sharing)

### Quarkus version or git rev

2.10.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

------------------------------------------------------------ Gradle 7.4.2 ------------------------------------------------------------   Build time:   2022-03-31 15:25:29 UTC Revision:     540473b8118064efcc264694cbcaa4b677f61041   Kotlin:       1.5.31 Groovy:       3.0.9 Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021 JVM:          11.0.14 (GraalVM Community 11.0.14+9-jvmci-22.0-b05) OS:           Windows 10 10.0 amd64

**Comment by geoand** [comment]
@zakkak @galderz this looks like a GraalVM bug to me. WDYT?

**Comment by geoand** [comment]
@martinhutton I would also encourage you to check out SQLLite extension in the [Quarkiverse](https://quarkiverse.github.io/quarkiverse-docs/quarkus-jdbc-sqlite/dev/index.html) as currently you are not using it so the application is rather unlikely to work in native mode.

**Comment by galderz** [comment]
It would seem you're using a custom base image for building native images?

```
quarkus.native.builder-image=graalbase
```

Also, looking at the logs above seems like you're on 22.0:

```
 Version info: 'GraalVM 22.0.0.2 Java 11 CE'
```

Try commenting the builder image property and see if it works? It would appear to be fixed in 22.1 (see https://github.com/oracle/graal/issues/4200 and linked issues/commits)

**Comment by gsmet** [comment]
If GraalVM is failing with a NPE, it's a GraalVM bug, not a Quarkus one, closing.
